packages/flutter/lib/src/material/slider.dart

```diff
@@ -34,6 +34,45 @@ typedef PaintValueIndicator = void Function(PaintingContext context, Offset offs
 
 enum _SliderType { material, adaptive }
 
+/// Possible ways a user can interact with a [Slider].
+enum SliderInteraction {
+  /// Allows the user to interact with a [Slider] by tapping or sliding anywhere
+  /// on the track.
+  ///
+  /// Essentially all possible interactions are allowed.
+  ///
+  /// This is different from [SliderInteraction.slideOnTrack] as when you try
+  /// to slide anywhere other than the thumb, the thumb will move to the first
+  /// point of contact.
+  tapAndSlide,
+  /// Allows the user to interact with a [Slider] by only tapping anywhere on
+  /// the track.
+  ///
+  /// Sliding interaction is ignored.
+  tapOnly,
+  /// Allows the user to interact with a [Slider] only by sliding anywhere on
+  /// the track.
+  ///
+  /// Tapping interaction is ignored.
+  slideOnTrack,
+  /// Allows the user to interact with a [Slider] only by sliding the thumb.
+  ///
+  /// Taping and sliding interactions on the track are ignored.
+  slideThumb,
+  /// Disallows the user to interact with a [Slider] in any way.
+  ///
+  /// This is different from setting [Slider.onChanged] to null as the [Slider]
+  /// does not look disabled.
+  none;
```

Comment:
Why do we need `none`?

Reply:
okay yeah that makes sense. i'll remove it.